monai/losses/shape.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from itertools import product
+from typing import Callable, Optional, Union
+
+import numpy as np
+import torch
+from scipy import ndimage
```

Comment:
please use `distance_transform_edt, _ = optional_import("scipy.ndimage.morphology", name="distance_transform_edt")` (example https://github.com/Project-MONAI/MONAI/blob/5c15138f484ba027b90fd12f7038a519f740d213/monai/metrics/utils.py#L22) because the project doesn't assume the user environment has scipy installed.